src/librustc_mir/interpret/intern.rs

```diff
@@ -45,6 +45,8 @@ struct InternVisitor<'rt, 'mir, 'tcx, M: CompileTimeMachine<'mir, 'tcx>> {
     /// despite the nested mutable reference!
     /// The field gets updated when an `UnsafeCell` is encountered.
     mutability: Mutability,
+
+    ignore_interior_mut_in_const_validation: bool,
```

Comment:
Add some docs linking this to the `mir::Body` flag

Reply:
/// This flag is to avoid triggering UnsafeCells are not allowed behind references in constants
    /// for promoteds.
    /// It's a copy of `mir::Body`'s ignore_interior_mut_in_const_validation field